File: c/455.分发饼干.c
/*
 * @lc app=leetcode.cn id=455 lang=c
 *
 * [455] 分发饼干
 */

// @lc code=start
int cmp(int *a, int *b){
    return *a - *b;
}

int findContentChildren(int* g, int gSize, int* s, int sSize) {
    qsort(g,gSize,sizeof(int),cmp); 
    qsort(s,sSize,sizeof(int),cmp); 

    int i = 0;
    for(int j = 0;i < gSize && j < sSize;j++)
    {
        if (g[i] <= s[j])
        {
            i++;
        }
    }
    return i;
}
// @lc code=end

File: rs/455.分发饼干.rs
/*
 * @lc app=leetcode.cn id=455 lang=rust
 *
 * [455] 分发饼干
 */

// @lc code=start
impl Solution {
    pub fn find_content_children(g: Vec<i32>, s: Vec<i32>) -> i32 {
        let (mut g,mut s) = (g,s);
        g.sort();
        s.sort();

        let mut i = 0;
        let mut j = 0;
        while(i < g.len() && j < s.len()){
            if g[i] <= s[j] {
                i += 1;
            }
            j += 1;
        }

        i as i32
    }
}
// @lc code=end

